### ISSUE TITLE
gcr.io/google_containers/pause-amd64:3.0 needs version update to support shared pids

### ISSUE THREAD

**Issue by Ninja3047** [kubernetes/kubernetes#50865]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
pause does not reap zombie processes using the default pause container provided by gcr.io/google_containers/pause-amd64:3.0

**What you expected to happen**:
gcr.io/google_containers/pause-amd64:3.0 should include the up to date pause code to reap zombie processes when using docker >= 1.13.1 found here

https://github.com/kubernetes/kubernetes/blob/master/build/pause/pause.c

**How to reproduce it (as minimally and precisely as possible)**:
```
kubectl create -f https://k8s.io/docs/tasks/debug-application-cluster/shell-demo.yaml
kubectl exec -it shell-demo -- /bin/bash
# apt update
# apt install curl gcc procps
# curl https://raw.githubusercontent.com/kubernetes/kubernetes/master/build/pause/orphan.c
# gcc orphan.c
# ./a.out
# ps ax
```
a.out will leave a zombie process that is not cleaned up

**Anything else we need to know?**:

Building the pause container from master solves this issue

Related:
https://github.com/kubernetes/kubernetes/pull/45236
https://github.com/kubernetes/kubernetes/pull/36853
https://github.com/kubernetes/kubernetes/issues/1615

**Environment**:
- Kubernetes version (use `kubectl version`): 1.7.0+
- Cloud provider or hardware configuration**: N/A
- OS (e.g. from /etc/os-release): CentOS 7
- Kernel (e.g. `uname -a`):  4.10.13+
- Install tools:
- Others: docker >= 1.13.1

**Comment by Ninja3047** [comment]
@kubernetes/sig-node-bugs

**Comment by sampsonhuo** [comment]
i get a simliar problem when i run crond service, every day it makes some  [anacron] <defunct> zombie progresses

UID        PID  PPID  C STIME TTY          TIME CMD
root         1     0  0 Nov14 ?        00:00:00 /pause
root         7     0  0 Nov14 ?        00:00:00 /usr/local/bin/dumb-init /usr/lo
root        13     7  0 Nov14 ?        00:00:00 nginx: master process /usr/local
nobody      21    13  0 Nov14 ?        00:00:01 nginx: worker process
nobody      22    13  0 Nov14 ?        00:00:07 nginx: worker process
nobody      23    13  0 Nov14 ?        00:00:07 nginx: worker process
nobody      24    13  0 Nov14 ?        00:00:05 nginx: worker process
root        77     1  0 Nov14 ?        00:00:00 [anacron] <defunct>
root     23276     1  0 00:01 ?        00:00:00 [anacron] <defunct>
root     23342     1  0 01:01 ?        00:00:00 [anacron] <defunct>
root     23408     1  0 02:01 ?        00:00:00 [anacron] <defunct>
root     23474     1  0 03:01 ?        00:00:00 [anacron] <defunct>
root     24286     0  0 11:49 ?        00:00:00 /bin/bash
root     24326     0  1 19:15 ?        00:00:00 /bin/bash
root     24347 24326  0 19:15 ?        00:00:00 ps -ef

**Comment by sampsonhuo** [comment]
i get a simliar problem when i run crond service, every day it makes some [anacron] zombie progresses

UID PID PPID C STIME TTY TIME CMD
root 1 0 0 Nov14 ? 00:00:00 /pause
root 7 0 0 Nov14 ? 00:00:00 /usr/local/bin/dumb-init /usr/lo
root 13 7 0 Nov14 ? 00:00:00 nginx: master process /usr/local
nobody 21 13 0 Nov14 ? 00:00:01 nginx: worker process
nobody 22 13 0 Nov14 ? 00:00:07 nginx: worker process
nobody 23 13 0 Nov14 ? 00:00:07 nginx: worker process
nobody 24 13 0 Nov14 ? 00:00:05 nginx: worker process
root 77 1 0 Nov14 ? 00:00:00 [anacron] 
root 23276 1 0 00:01 ? 00:00:00 [anacron]  <defunct>
root 23342 1 0 01:01 ? 00:00:00 [anacron]  <defunct>
root 23408 1 0 02:01 ? 00:00:00 [anacron]  <defunct>
root 23474 1 0 03:01 ? 00:00:00 [anacron]  <defunct>
root 24286 0 0 11:49 ? 00:00:00 /bin/bash
root 24326 0 1 19:15 ? 00:00:00 /bin/bash
root 24347 24326 0 19:15 ? 00:00:00 ps -ef

![image](https://user-images.githubusercontent.com/20149979/32833510-13cb01dc-ca3a-11e7-98e7-03563cb4ad26.png)

**Comment by ixdy** [comment]
Do we know why this was never resolved / #36853 was never deployed?

**Comment by mrballcb** [comment]
I just fought through this issue with a rabbitmq helm chart.  I had to custom build a pause container from master to fix it.

The rabbitmq helm chart uses `rabbitmqctl status` for its liveness checks.  It's a thin wrapper around a call to the rabbitmq-server daemon, which is not PID 1 (/pause is PID 1).  The rabbitmq-server process forks some children to do the work and then they exit.  Since the pause-amd64:3.0 is not reaping, those processes stick around as zombies.  The liveness check runs every 5 seconds.  After about 4 hours, there are about 27K zombies and the node begins failing to launch new containers because it's starved for resources.  The pause container built from current master fixes that behavior, no zombies at all in the container.

https://github.com/kubernetes/charts/issues/2989

**Comment by verb** [comment]
The plan is to build a new version of pause-amd64 and then update the image referenced by Kubernetes. This means the default image will change in the next Kubernetes release.

We can make the new image available once the freeze lifts and we can merge #56762.

**Comment by macropin** [comment]
Looks like this made it into 1.10 https://github.com/kubernetes/kubernetes/blob/release-1.10/build/pause/CHANGELOG.md
